From 96867fc8af2103d07682c769942dc6532b6b99f9 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Tue, 30 Oct 2018 08:50:18 +0200
Subject: [PATCH 0740/1051] arm64: dts: marvell: armada-37xx: add utmi phy
 definition

In addition, connect the UTMI PHY to the USB2 controller.

This fixes USB2 functionality on A3700 devices.

Change-Id: I579774ac06885edad9289f7669b11e169a2c4561
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60601
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../arm64/boot/dts/marvell/armada-3720-db.dts | 22 +++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi  |  9 ++++++++
 2 files changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index e8cbc1fa0d07..5dfa38e0ae9b 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -44,6 +44,21 @@
 		vcc-supply = <&exp_usb3_vbus>;
 	};
 
+	exp_usb2_vbus: usb2-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb2-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&gpio_exp 0 GPIO_ACTIVE_HIGH>;
+	};
+
+	usb2_phy: usb2-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb2_vbus>;
+	};
+
 	vcc_sd_reg1: regulator {
 		compatible = "regulator-gpio";
 		regulator-name = "vcc_sd1";
@@ -202,6 +217,13 @@
 /* CON27(V2.0)/CON29(V1.4) */
 &usb2 {
 	status = "okay";
+	usb-phy = <&usb2_phy>;
+	phys = <&utmi_usb2>;
+	phy-names = "usb";
+};
+
+&utmi_usb2 {
+	status = "okay";
 };
 
 /* CON29(V2.0)/CON31(V1.4) */
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 2def182decc4..a03efc7b787d 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -9,6 +9,7 @@
  */
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/phy/phy-utmi-mvebu.h>
 
 / {
 	model = "Marvell Armada 37xx SoC";
@@ -333,6 +334,14 @@
 				status = "disabled";
 			};
 
+			utmi_usb2: utmi@5f000 {
+				compatible = "marvell,armada-3700-utmi-phy";
+				reg = <0x5f000 0x1000>;
+				utmi-port = <UTMI_PHY_TO_USB2_HOST0>;
+				#phy-cells = <0>;
+				status = "disabled";
+			};
+
 			xor@60900 {
 				compatible = "marvell,armada-3700-xor";
 				reg = <0x60900 0x100>,
-- 
2.17.1

